crates/guest-rust/src/lib.rs

```diff
@@ -250,7 +250,18 @@ impl<T: WasmResource> Resource<T> {
     where
         T: RustResource,
     {
-        let _ = Box::from_raw(rep as *mut T);
+        let _ = Box::from_raw(rep as *mut Option<T>);
```

Comment:
Would you be up for a small refactoring of this module? If so I think it'd be good to have something like 

```rust
type RawRep<T> = Option<T>;
```

or something like that where all of these casts and various "raw" methods would work with `RawRep<T>`. That way future changes to `RawRep<T>` would help trigger compilation errors where appropriate. Basically I think it'd be good to centralize the representation of `Option<T>` if possible.